Sources/RInAppMessaging/CampaignDispatcher.swift

```diff
@@ -100,28 +104,46 @@ internal class CampaignDispatcher: CampaignDispatcherType, TaskSchedulable {
             dispatchNext()
             return
         }
+        fetchCampaignImagesAndDisplay(campaign: campaign)
+    }
+    
+    func fetchCampaignImagesAndDisplay(campaign: Campaign) {
+        guard let resImgUrlString = campaign.data.messagePayload.resource.imageUrl,
+              let resImgUrl = URL(string: resImgUrlString) else {
+            // If no image expected, just display the message.
+            displayCampaign(campaign)
+            return
+        }
 
-        // fetch from imageUrl, display if successful, skip on error
-        if let resImgUrlString = campaign.data.messagePayload.resource.imageUrl, let resImgUrl = URL(string: resImgUrlString) {
-            data(from: resImgUrl) { imgBlob in
-                self.dispatchQueue.async {
-                    guard let imgBlob = imgBlob else {
-                        self.dispatchNext()
-                        return
-                    }
-                    self.displayCampaign(campaign, imageBlob: imgBlob)
+        fetchImage(from: resImgUrl, for: campaign)
```

Comment:
it will always have a value for carousel type campaign. The user has to upload an image and make it an image only campaign . Then the image links will be processed for carousel

Reply:
Are we using that image anywhere? if not, why should we download it?